### PR TITLE
Add extension point on filter, gallery and product summary

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?
Move the gallery, product summary and filter navigator to extension points, instead of directly referencing the components in the search result.

#### What problem is this solving?
You couldn't fully customize the search before, like changing the product summary of the gallery, or the whole gallery component.

#### How should this be manually tested?
[Workspace](https://lucas--storecomponents.myvtex.com/eletronicos/smartphones).

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -27,7 +27,6 @@
   "dependencies": {
     "vtex.styleguide": "6.x",
     "vtex.product-summary": "1.x",
```
Comment (claudivanfilho) [inline manifest.json]: tem que remover essa dependência aqui tb não?

Comment (claudivanfilho) [inline manifest.json]: Tem que remover essa dependência aqui tb não?

Comment (lucasecdb) [reply manifest.json]: Não, no `index` ela ta sendo usada pra pegar o schema.

Comment (claudivanfilho) [reply manifest.json]: Opa, então isso é uma issue certo? Esse esquema tem que ser capturado dinamicamente, assim como ele é passado no pages.json. Vou aceitar o PR, mas registra isso lá no `Ideas`do trello pra q a gente possa discutir depois como vamos resolver. :)

File: manifest.json [hunk 2]
```diff
@@ -27,7 +27,6 @@
...
-    "vtex.store-graphql": "2.x",
-    "vtex.breadcrumb": "0.x"
+    "vtex.store-graphql": "2.x"
```
Comment (brunojdo) [inline manifest.json]: The search-result make any query or mutation ?

**Pull request merged**